src/tensors/linalg.jl

```diff
@@ -83,10 +90,17 @@ error will be thrown.
 
 See also [`unitary`](@ref) when `InnerProductStyle(cod) === EuclideanProduct()`.
 """
-function isomorphism(::Type{A}, V::TensorMapSpace{S,N₁,N₂}) where {A<:MatOrNumber,S,N₁,N₂}
+function isomorphism(::Type{A}, V::TensorMapSpace{S,N₁,N₂}) where {A<:VecOrNumber,S,N₁,N₂}
     codomain(V) ≅ domain(V) ||
         throw(SpaceMismatch("codomain and domain are not isomorphic: $V"))
-    t = tensormaptype(S, N₁, N₂, A)(undef, codomain(V), domain(V))
+    if A <: Number
+        t = TensorMap{A}(undef, V)
+    elseif A <: DenseVector
+        T = scalartype(A)
+        t = TensorMap{T,S,N₁,N₂,A}(undef, V)
+    else
+        throw(ArgumentError("`isomorphism` only supports Number or DenseVector subtypes as first argument"))
```

Comment:
I think this is precluded by the function signature

Reply:
Yeah, I first thought of getting rid of the type restriction all together, which is currently done at some places, so I guess we have to be consistent. I think I would actually like the error message rather than a generic MethodError, so I think I will get rid of the type restriction.